### PR TITLE
testing: consolidate list of modules into a single place

### PR DESCRIPTION
**Pull request by eliben**
Strawman that only updates listdeps.sh for now, just to test the idea. WDYT?

The reason I'm placing it in a .sh file that I `source` is that we can add comments to it explaining what it does and also makes the "user" scripts cleaner.

A similar script can be added later to generate the vscode stuff

Updates #1883

### CODE REVIEW

**Comment by zombiezen** [comment]
This looks nice. One idea: you could enter the directories line by line and still have comments by reading it using:

```bash
mapfile -t ALLMODULES < <(sed -e '/^#/d' path/to/file.txt)
```

**Comment by zombiezen** [comment]
Amending for blank lines:

```bash
mapfile -t ALLMODULES < <(sed -e '/^#/d' -e '/^$/d' path/to/file.txt)
```

**Comment by eliben** [comment]
Thanks. Changed to data file and using `sed | while read` to read now. PTAL

File: allmodules [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# This file lists all the modules in our repository; it's used by scripts that
+# have to perform operations for each module.
+#
+# Any line that doesn't begin with a '#' character and isn't empty is treated
+# as a path that has a module in it.
```
Comment (zombiezen) [inline allmodules]:
```suggestion
# as a path relative to the top of the repository that has a module in it.
```

File: allmodules [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+.
+./internal/cmd/gocdk
```
Comment (zombiezen) [inline allmodules]: nit: these don't need the `./` prefix.

Comment (eliben) [reply allmodules]: Done

**Pull request merged**